fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -1207,6 +1206,7 @@ public void finishTransaction(long transactionId, Set<Long> errorReplicaIds) thr
             finishSpan.end();
         }
 
+        transactionState.afterStateTransform(TransactionStatus.VISIBLE, txnOperated);
```

Comment:
This bug does not exist in the main branch. Because the lock is not needed here.
![image](https://github.com/user-attachments/assets/84f16c98-161a-462c-a804-f55704978583)